src/features/devtools/components/ExpressionPlayground/ExpressionPlayground.tsx

```diff
@@ -76,8 +76,11 @@ export const ExpressionPlayground = () => {
 
   const traversalSelector = useNodeTraversalSelector();
 
-  const componentOptions = useNodeTraversal((t) =>
-    t.allNodes().map((n) => ({ label: n.id, value: `${n.page.pageKey}|${n.id}` })),
+  const componentOptions = NodesInternal.useShallowSelector((state) =>
+    Object.values(state.nodeData).map((nodeData) => ({
+      label: nodeData.layout.id,
+      value: `${nodeData.pageKey}|${nodeData.layout.id}`,
+    })),
```

Comment:
A shallow selector will not do anything in this case no? You are returning a list, so the elements of the list are shallow-compared for equality, but you are creating new objects to populate the list every time so it will always fail the equality check on the first element

Reply:
Absolutely true! Sloppy mistake on my part. :face_with_peeking_eye: